src/preprocessing/passes/bv_ackermann.cpp

```diff
@@ -54,42 +93,82 @@ void storeFunction(
   NodeSet& set = fun_to_args[func];
```

Comment:
`func_to_args` is a `FunctionToArgsMap`, which is defined as `std::unordered_map<Node, NodeSet, NodeHashFunction>`. However, all nodes are inserted as a mapping of TNode -> Set(TNode). It seems to me that this could rather be a map from TNode to TNodeSet. But if you change this, double check if all inserted nodes are indeed TNodes, else you'll run into trouble.